### ISSUE TITLE
[Clang-Tidy] Create readability-logic-missing-parentheses check

### ISSUE THREAD

**Issue by PiotrZSL** [llvm/llvm-project#80852]
I'm adding this here so, i wouldn't forget.
Create check that detect missing () around logical expressions when operators of different priority are used.
```
    x && y || z;
```
Should be:
```
    (x && y) || z;
```
Main purpose is to explicitly show evaluation order.
Add option to ignore short expressions, as this will be way useful when many operators are used in single `if` or where expressions are long, like some function calls.

**Comment by chrchr-github** [comment]
Is this different from `clang-diagnostic-logical-op-parentheses`?

**Comment by PiotrZSL** [comment]
> Is this different from `clang-diagnostic-logical-op-parentheses`?

Yes, you right this warning covers it. Somehow I didn't notice it.
